pkg/apply/prune/prune.go

```diff
@@ -241,6 +241,20 @@ func (po *PruneOptions) Prune(currentObjects []*resource.Info, eventChannel chan
 		if po.currentUids.Has(uid) {
 			continue
 		}
+		// Handle lifecycle directive preventing deletion.
+		if preventDeleteAnnotation(metadata.GetAnnotations()) {
+			eventChannel <- event.Event{
+				Type: event.ApplyType,
```

Comment:
Using the `Apply` event type doesn't seem right here. The annotation is about pruning, not apply.

Reply:
Added new `PruneEventType` called `PruneEventSkipped`. My initial pass at printing this looks like:

```
$ kapply apply ~/testdata/testdata-2
configmap/inventory-aa61e798 created
namespace/test-namespace unchanged
pod/pod-b unchanged
pod/pod-c unchanged
pod/pod-d created
5 resource(s) applied. 2 created, 3 unchanged, 0 configured
pod/pod-a pruned skipped
configmap/inventory-1e5824fb pruned
1 resource(s) pruned
```

Let me know what you think.